### ISSUE TITLE
Cannot Extend widget.Filetree

### ISSUE THREAD

**Issue by Intyflay** [fyne-io/fyne-x#45]
The following code results in a thin blank bar with no content.
```
type customtree struct {
	xwidget.FileTree
}

func NewCustomTree(root fyne.URI) *customtree {
	tree := &customtree{}
	tree.ExtendBaseWidget(tree)
	tree.Root = root.String()
	return tree
}
```

**Comment by Jacalz** [comment]
Yeah that’s because the constructor unfortunately sets up pretty much all of the logic for the widget: https://github.com/fyne-io/fyne-x/blob/27fab34c671cc883bc96a095924c2d67067dc587/widget/filetree.go#L25. This could be fixed by moving some of the logic to the CreateRenderer() method.
